packages/controller-utils/src/util.ts

```diff
@@ -539,3 +539,40 @@ function logOrRethrowError(error: any, codesToCatch: number[] = []) {
     throw error;
   }
 }
+
+/**
+ * Resolve all pending promises.
+ * This method is used for async tests that use fake timers.
+ * See https://stackoverflow.com/a/58716087 and https://jestjs.io/docs/timer-mocks.
+ */
+const flushPromises = () => {
```

Comment:
kinda rough that this ends up becoming part of non-dev deps code. Do we have a package that we can put code like this which is  only ever imported as a devDependency?

Reply:
yeah I guess that would be ideal... I don't believe we have a repo like that ATM... @mcmire @Gudahtt ?

Reply:
Fortunately its very few lines in any case.

Reply:
Ah! So we don't have to share this by putting this in a package. We have a way of sharing helpers across the tests for multiple packages. There's a `tests/` directory in the root that we use for this purpose. We already have a couple of files `fake-block-tracker.ts` and `mock-network.ts`, but we could also put a `helpers.ts` file in there to hold helper functions and then import it from any test file. Note that we may have to modify the `rootDir` of the package in order to do this. You can see this being done for NetworkController: [here](https://github.com/MetaMask/core/blob/1d0afa35421e3e498c50f4c151b74b43de52de33/packages/network-controller/tests/NetworkController.test.ts#L17) we're importing `fake-block-tracker` and [here](https://github.com/MetaMask/core/blob/1d0afa35421e3e498c50f4c151b74b43de52de33/packages/network-controller/tsconfig.json#L5) we've set `rootDir` to `../..` to instruct TypeScript that it's okay to look outside this package for other files, and we've also added `../../tests` to the list of directories it should look in.

Reply:
oh lol @mcmire  I didn't see you're comment when I wrote mine